### ISSUE TITLE
'first argument must be a string or Buffer'

### ISSUE THREAD

**Issue by Subtixx** [emoose/DyingLightDocketServer#1]
```_http_outgoing.js:542
    throw new TypeError('first argument must be a string or Buffer');
    ^

TypeError: first argument must be a string or Buffer
    at ServerResponse.OutgoingMessage.end (_http_outgoing.js:542:11)
    at ClientRequest.<anonymous> (nexe.js:155:18)
    at emitOne (events.js:90:13)
    at ClientRequest.emit (events.js:182:7)
    at TLSSocket.socketErrorListener (_http_client.js:295:9)
    at emitOne (events.js:90:13)
    at TLSSocket.emit (events.js:182:7)
    at emitErrorNT (net.js:1250:8)
    at _combinedTickCallback (node.js:382:13)
    at process._tickCallback (node.js:406:11)
```

Possible fix (untested):

```js
req.on('error', function(err)
    {
        response.writeHead(404, {'Content-Type': 'application/json'});
        response.end(err);
    });
```

to

```js
req.on('error', function(err)
    {
        response.writeHead(404, {'Content-Type': 'application/json'});
        response.end(err.toString());
    });
```

**Comment by Subtixx** [comment]
Also just noticed.. Normal request dont get through. Always:
```
SyntaxError: JSON.parse: unexpected character at line 1 column 1 of the JSON data
Error: connect ECONNREFUSED 52.71.70.189:443
```

Possible that the ip changed to: `52.1.125.101` but II only get "Not found"

**Comment by emoose** [comment]
Ah that's unfortunate, it's been years since I worked on this so lots has probably changed since then...

If you want to try fixing it up you could try installing [Fiddler](https://www.telerik.com/fiddler), set it up to capture/decrypt HTTPS (Tools -> Options -> HTTPS), and then run the game and see what it connects to, I can't remember if Fiddler worked by itself or not though, you might need to use something like Proxifier to force the game to connect through Fiddler first.

EDIT: it looks like the IP might be "34.199.102.221", at least pls.dyinglightgame.com seems to use that IP, you might still want to check with Fiddler though since other things could have been changed...

**Comment by Subtixx** [comment]
Yea, but the pls is hosted on amazonaws so the IP can be very different for people living in the us, eu etc. You could also use the amazonaws URL (Which would be `lb-production-us-east-1-1658530636.us-east-1.elb.amazonaws.com`) that you get via nslookup. Problem is it always answers with not found. I'll try the proxifier / fiddler method but I'm playing the game on the ps4. The dockets worked even though the server can't relay Requests, which is really cool (and weird since for that to work the device [in this case the game] has to trust the self-signed certificate which it does) but after 1 request the game thinks the servers are down so it doesn't make another one.

**Comment by Subtixx** [comment]
Hello, I'm back playing the game again, but this time on PC. Replacing now the fixed ip with `lb-production-us-east-1-1658530636.us-east-1.elb.amazonaws.com` works.
